node/src/components/consensus/protocols/highway.rs

```diff
@@ -582,9 +584,66 @@ where
                     }
                 }
             }
+            Ok(HighwayMessage::RequestLatestState(panorama)) => {
+                trace!("received a request for the latest state");
+                let state = self.highway.state();
+
+                let create_message =
+                    |observations: (&Observation<C>, &Observation<C>)| match observations {
+                        (Observation::None, Observation::None)
+                        | (Observation::Faulty, _)
+                        | (_, Observation::Faulty) => None,
+
+                        (Observation::None, Observation::Correct(hash)) => {
+                            Some(HighwayMessage::RequestDependency(Dependency::Unit(*hash)))
+                        }
+
+                        (Observation::Correct(hash), Observation::None) => state
+                            .wire_unit(hash, *self.highway.instance_id())
+                            .map(|swu| HighwayMessage::NewVertex(Vertex::Unit(swu))),
+
+                        (Observation::Correct(our_hash), Observation::Correct(their_hash)) => {
+                            if state.has_unit(their_hash)
+                                && state.panorama().sees_correct(state, their_hash)
+                            {
+                                state
+                                    .wire_unit(our_hash, *self.highway.instance_id())
+                                    .map(|swu| HighwayMessage::NewVertex(Vertex::Unit(swu)))
+                            } else if !state.has_unit(their_hash) {
+                                Some(HighwayMessage::RequestDependency(Dependency::Unit(
+                                    *their_hash,
+                                )))
+                            } else {
+                                None
+                            }
+                        }
+                    };
+
+                state
+                    .panorama()
+                    .iter()
+                    .zip(&panorama)
+                    .filter_map(create_message)
+                    .map(|msg| {
+                        let serialized_msg =
+                            bincode::serialize(&msg).expect("should serialize message");
+                        ProtocolOutcome::CreatedTargetedMessage(serialized_msg, sender.clone())
+                    })
+                    .collect()
+            }
         }
     }
 
+    fn handle_new_peer(&mut self, peer_id: I) -> Vec<ProtocolOutcome<I, C>> {
+        trace!(?peer_id, "connected to a new peer");
```

Comment:
I would remove this trace. It's logged elsewhere already.

Reply:
But this log tells us that consensus is actually handling this. I'd prefer to leave it, unless you feel strongly about it. Might help debugging in my opinion, though.